bacom-blog/pull-quote-update.js

```diff
@@ -0,0 +1,122 @@
+import { getLeaf, getNodesByType } from '../utils/mdast-utils.js';
+
+function noAuthorQuoteRow(quoteObj = '') {
+    const emptyNode = '{"type":"text","value":""}';
+
+    return `{"type":"gtRow","children":[{"type":"gtCell","children":[{"type":"heading","depth":3,"children":[${quoteObj || emptyNode}]}],"valign":"middle"}]}`;
+}
+
+function authorQuoteRow(quoteObj = '', authorObj = '', attributionObj = '') {
+    const emptyNode = '{"type":"text","value":""}';
+
+    return `{"type":"gtRow","children":[{"type":"gtCell","children":[{"type":"paragraph","children":[${quoteObj || emptyNode}]},{"type":"paragraph","children":[${authorObj || emptyNode}]},{"type":"paragraph","children":[${attributionObj || emptyNode}]}],"valign":"middle"}]}`
+}
+
+function splitQuoteAttribution(node, replacement) {
+    const attribution = node?.value;
+    if (!attribution) {
+        console.log('Issue with quote attribution');
+        return;
+    }
+    const splitAttr = attribution.split(',');
+
+    if (splitAttr.length === 1) {
+        replacement.author = JSON.stringify({type: 'text', value: splitAttr[0]});
+        return;
+    }
+
+    let [author, ...attr] = splitAttr;
+    attr = attr.join(',');
+
+    replacement.author = JSON.stringify({type: 'text', value: author});
+    replacement.company = JSON.stringify({type: 'text', value: attr});;
+}
+export const QUOTE_BLOCK_NAME = 'quote (borders, align left)';
+/**
+ * 
+ * @param {*} mdast
+ * 
+ * Takes the current mdast, finds all instances of pull quote, and changes them to quote. Likewise modifies
+ * content into formats expected by Milo. 
+ *  
+ */
+export async function convertPullQuote(mdast) {
+    const quoteMap = mdast.children.reduce((rdx, child, index) => {
+        const header = getLeaf(child, 'text');
+
+        if (!header) return rdx;
+        if (typeof header.value !== 'string') return rdx;
+        if (!header?.value.toLowerCase().includes("quote")) return rdx;
+
+        rdx.push({
+            headText: header.value,
+            index: index
+        });
+        return rdx;
+    }, [])
```

Comment:
I don't see headText being used, just index, filter would be better here than reduce

Reply:
True on not using the value, removed it. Filter though returns the current iterable based on a true or false return from the callback. I need to capture the index based on the logic within the callback, so I went with reduce to create an array of those. 

Perhaps there is some greater refactoring that we could do to utilize filter? 

I did modify this to shorten it, using your's and Megan's comments.